src/inst.c

```diff
@@ -256,15 +260,23 @@ LRESULT CALLBACK InstrumentsWndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM
 	case WM_COMMAND: {
 		WORD id = LOWORD(wParam), action = HIWORD(wParam);
 		switch (id) {
-		case IDC_MIDIINCOMBO:
-			if (action == CBN_SELCHANGE) {
-				midiDevice = SendMessage((HWND)lParam, CB_GETCURSEL, 0, 0) - 1;
-				closeMidiInDevice();
-				openMidiInDevice(midiDevice, MidiInProc);
-			} else if (action == CBN_CLOSEUP) {
-				SetFocus(instlist);
-			}
-			break;
+			case IDC_MIDIINCOMBO:
+				if (action == CBN_SELCHANGE) {
+					midiDevice = SendMessage((HWND)lParam, CB_GETCURSEL, 0, 0) - 1;
+					closeMidiInDevice();
+					openMidiInDevice(midiDevice, MidiInProc);
+				} else if (action == CBN_CLOSEUP) {
+					SetFocus(instlist);
+				}
+				break;
+			case IDC_INSTLIST:
+				if (action == LBN_SELCHANGE) {
+					int sel = SendMessage((HWND)lParam, LB_GETCURSEL, 0, 0);
+					struct channel_state *c = &state.chan[0];
+					set_inst(&state, c, valid_insts[sel]);
+					format_status(0, "ADSR: %02d/15  %d/7  %d/7  %02d/31", 0xF - (c->inst_adsr1 & 0xF), (c->inst_adsr1 >> 4) & 0x7, (c->inst_adsr2 >> 5) & 7, c->inst_adsr2 & 0x1F);
```

Comment:
Most resources for SNES envelopes out there leave attack as "higher = ramps up faster," so it might be better to drop the 0xF - attack part and display it raw. Your choice, though.

This doesn't handle GAIN either (when `(c->inst_adsr1 & 0x80) == 0`) but that's not relevant to EarthBound.

Reply:
Compared to actual ADSR this is weird, but ok